api/src/database/tamagotchi/getByID.ts

```diff
@@ -0,0 +1,23 @@
+import getDatabaseConnection from '../../shared/database';
+import { Tamagotchi } from '../../types/Tamagotchi';
+
+type TamagotchiGetByIDOptions = Pick<Tamagotchi, 'id'>;
+
+const tamagotchiGetByID = ({ id }: TamagotchiGetByIDOptions) =>
+  new Promise<Tamagotchi | null>((resolve, reject) => {
+    const databaseConnection = getDatabaseConnection();
+    databaseConnection.execute<Tamagotchi[][]>(
+      { sql: 'CALL tamagotchi_get_by_id()', values: [] },
```

Comment:
Does this file even get used?

```suggestion
      { sql: 'CALL tamagotchi_get_by_id(?)', values: [id] },
```